contracts/Balancer/UniswapMarketMaker.sol

```diff
@@ -181,65 +180,45 @@ contract UniswapMarketMaker is IERC721Receiver, Stabilizer {
                 deadline: block.timestamp
             })
         );
-
         _collect(tokenId);
     }
 
     /**
-     * @notice Collects the fees associated with provided liquidity
-     * @dev The contract must hold the erc721 token before it can collect fees
-     */
-    function collect()
-        external
-        onlyBorrower
-        whenNotPaused
-        nonReentrant
-        isMinted
-    {
-        _collect(tokenId);
-    }
-
-    /**
-     * @notice Burn NFT
+     * @notice Increases liquidity in the current range
+     * @dev Pool must be initialized already to add liquidity
+     * @param usdxAmount USDX Amount of asset to be deposited
+     * @param sweepAmount Sweep Amount of asset to be deposited
+     * @param usdxMinIn Min USDX amount to be used for liquidity.
+     * @param sweepMinIn Min Sweep amount to be used for liquidity.
      */
-    function burnNFT() external onlyBorrower nonReentrant isMinted {
-        nonfungiblePositionManager.burn(tokenId);
-        tokenId = 0;
-    }
-
-    function initPool(
-        uint256 usdxAmount,
-        uint256 sweepAmount,
-        uint256 usdxMinAmount,
-        uint256 sweepMinAmount
-    )
-        external onlyBorrower nonReentrant
-        returns (
-            uint256 _tokenId,
-            uint128 _liquidity,
-            uint256 _amount0,
-            uint256 _amount1
-        )
+    function lpTrade(uint256 usdxAmount, uint256 sweepAmount, uint256 usdxMinIn, uint256 sweepMinIn, uint256 spread)
+        external onlyBorrower whenNotPaused nonReentrant
     {
         address self = address(this);
-        if (sweep.isMintingAllowed()) {
-            _borrow(sweepAmount);
-        } else {
-            TransferHelper.safeTransferFrom(address(sweep), msg.sender, self, sweepAmount);
+        if(tradePosition > 0) _removePosition(tradePosition);
+        (uint256 usdxBalance, uint256 sweepBalance) = _balances();
+
+        if(sweepAmount > sweepBalance) {

```

Comment:
I would get rid of lines 201 to 211. My position is that balance should be in the contract before calling this function. Transfers in of usdc should happen in buySweep.